src/main/kotlin/no/nav/syfo/sykmelding/KafkaBehandlerSykmelding.kt

```diff
@@ -0,0 +1,88 @@
+package no.nav.syfo.behandler.kafka.sykmelding
+
+import no.nav.syfo.ApplicationState
+import no.nav.syfo.Environment
+import no.nav.syfo.database.database
+import no.nav.syfo.domain.*
+import no.nav.syfo.kafka.kafkaAivenConsumerConfig
+import no.nav.syfo.personoppgave.createPersonOppgave
+import no.nav.syfo.personoppgave.domain.PersonOppgaveType
+import no.nav.syfo.personoppgave.getPersonOppgaverByReferanseUuid
+import org.apache.kafka.clients.consumer.*
+import java.sql.Connection
+import java.time.*
+import java.util.*
+
+const val SYKMELDING_TOPIC = "teamsykmelding.ok-sykmelding"
+
+fun launchKafkaTaskSykmelding(
+    applicationState: ApplicationState,
+    environment: Environment,
+) {
+    val consumerProperties = kafkaAivenConsumerConfig<ReceivedSykmeldingDTO>(environment.kafka).apply {
+        this[ConsumerConfig.AUTO_OFFSET_RESET_CONFIG] = "latest"
+    }
+    val kafkaConsumerSykmelding = KafkaConsumer<String, ReceivedSykmeldingDTO>(consumerProperties)
+
+    kafkaConsumerSykmelding.subscribe(
+        listOf(SYKMELDING_TOPIC)
+    )
+    while (applicationState.ready) {
+        pollAndProcessSykmelding(
+            kafkaConsumerSykmelding = kafkaConsumerSykmelding,
+        )
+    }
+}
+
+fun pollAndProcessSykmelding(
+    kafkaConsumerSykmelding: KafkaConsumer<String, ReceivedSykmeldingDTO>,
+) {
+    val records = kafkaConsumerSykmelding.poll(Duration.ofMillis(1000))
+    if (records.count() > 0) {
+        processSykmelding(
+            consumerRecords = records,
+        )
+        kafkaConsumerSykmelding.commitSync()
+    }
+}
+
+fun processSykmelding(
+    consumerRecords: ConsumerRecords<String, ReceivedSykmeldingDTO>,
+) {
+    database.connection.use { connection ->
+        consumerRecords.forEach {
+            it.value()?.let { receivedSykmeldingDTO ->
+                COUNT_MOTTATT_SYKMELDING.increment()
+                if (
+                    receivedSykmeldingDTO.sykmelding.meldingTilNAV != null &&
+                    receivedSykmeldingDTO.sykmelding.meldingTilNAV.beskrivBistand != null
+                ) {
+                    createPersonoppgave(
+                        connection = connection,
+                        receivedSykmeldingDTO = receivedSykmeldingDTO,
+                    )
+                }
+            }
+        }
+        connection.commit()
+    }
+}
+
+private fun createPersonoppgave(
+    connection: Connection,
+    receivedSykmeldingDTO: ReceivedSykmeldingDTO,
+) {
+    val referanseUuid = UUID.fromString(receivedSykmeldingDTO.msgId)
```

Comment:
Har du sjekket at `msgId`-feltet kan gjenfinnes i responsen fra APIet vi bruker for å hente sykmeldinger i Modia?

Reply:
Tok den første å beste uuid'en. Men kan sjekke 👍

Reply:
Så lenge vi kan bruke UUID-en til å matche en sykmelding i frontend er vi good :+1:

Reply:
Kan se ut som det er bedre å bruke `receivedSykmeldingDTO.sykmelding.id`